side/src/main/java/com/ssafy/side/api/email/service/EmailService.java

```diff
@@ -0,0 +1,94 @@
+import com.ssafy.side.common.exception.ErrorMessage;
+import com.ssafy.side.common.exception.InternalServerException;
+import com.ssafy.side.common.util.RedisUtil;
+import jakarta.mail.internet.MimeMessage;
+import java.security.SecureRandom;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.stereotype.Service;
+import org.thymeleaf.context.Context;
+import org.thymeleaf.spring6.SpringTemplateEngine;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class EmailService {
+    private final JavaMailSender mailSender;
+    private final RedisUtil redisUtil;
+    private final SpringTemplateEngine templateEngine;
+    @Value("${spring.mail.username}")
+    private String emailAddress;
+    @Value("${spring.mail.personal}")
+    private String replacedEmailAddress;
+    @Value("${spring.mail.auth-code-expiration-millis}")
+    private long expirationMillis;
+
+    private final String upperCaseLetters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    private final String lowerCaseLetters = "abcdefghijklmnopqrstuvwxyz";
+    private final String numbers = "0123456789";
+
+    public EmailSendResponseDto sendVerificationEmail(String email) {
+        if (redisUtil.hasKey(email)) {
+            redisUtil.deleteValue(email);
+        }
+
+        Context context = new Context();
+        String verificationCode = generateVerificationCode();
+        context.setVariable("verificationCode", verificationCode);
+        String htmlContent = templateEngine.process("verification-email", context);
+
+        try {
+            MimeMessage message = mailSender.createMimeMessage();
+            MimeMessageHelper messageHelper = new MimeMessageHelper(message, true, "UTF-8");
+            messageHelper.setSubject("SANDBOX 이메일 인증");
+            messageHelper.setFrom(emailAddress, replacedEmailAddress);
+            messageHelper.setTo(email);
+            messageHelper.setText(htmlContent, true);
+            mailSender.send(message);
+
+            redisUtil.setValueWithExpiration(email, verificationCode, expirationMillis);
+
+            return new EmailSendResponseDto(true);
+        } catch (Exception e) {
+            log.error("Failed to send verification email to {}: {}", email, e.getMessage(), e);
+            throw new InternalServerException(ErrorMessage.ERR_EMAIL_FAILED_TO_SEND);
+        }
+    }
+
+    public String generateVerificationCode() {
+        String allCharacters = upperCaseLetters + lowerCaseLetters + numbers;
+
+        SecureRandom random = new SecureRandom();
+        StringBuilder code = new StringBuilder(6);
+
+        for (int i = 0; i < 6; i++) {
+            int index = random.nextInt(allCharacters.length());
+            code.append(allCharacters.charAt(index));
+        }
+
+        return code.toString();
```

Comment:
```suggestion
             return new SecureRandom().ints(6, 0, allCharacters.length())
                         .mapToObj(allCharacters::charAt)
                         .collect(StringBuilder::new, StringBuilder::append, StringBuilder::append)
                         .toString();

```

로 간단하게 정리할 수 있을 것 같아요 !

Reply:
단순한 반복 작업이라, Stream API를 사용하는 것보다 for-loop가 빠르다고 생각해서 for-loop를 사용했었습니다. 

https://sigridjin.medium.com/java-stream-api%EB%8A%94-%EC%99%9C-for-loop%EB%B3%B4%EB%8B%A4-%EB%8A%90%EB%A6%B4%EA%B9%8C-50dec4b9974b

그래도, 예슬님 말씀대로 간결하게 보이기 위해서 Stream API 사용해서 변경하겠습니다 !